fixtures/dom/src/components/fixtures/date-inputs/switch-date-test-case.js

```diff
@@ -0,0 +1,54 @@
+const React = window.React;
+
+/**
+ * This test case was originally provided by @richsoni,
+ * https://github.com/facebook/react/issues/8116
+ */
+class SwitchDateTestCase extends React.Component {
+  state = {
+    fullDate: false,
+    date: new Date().toISOString()
+  }
+
+  render() {
+    const attrs = this.inputAttrs();
+
+    return (
+      <div>
+        <p><b>{attrs.type}</b> input type ({attrs.value})</p>
+        <p>
+          <input type={attrs.type} value={attrs.value} onChange={this.onInputChange} />
+          <label><input type='checkbox' checked={this.state.fullDate} onChange={this.updateFullDate} /> Switch type</label>
+        </p>
+      </div>
+    );
+  }
+
+  inputAttrs() {
+    if (this.state.fullDate){
+      return {
+        type: 'datetime-local',
+        value: this.state.date.replace(/\..*Z/,''),
```

Comment:
This may not be relevant to this fixture, but by trimming these date strings we're changing the actual value of the date. For example:

```js
new Date("2017-07-12T14:28:58.590Z")
// Wed Jul 12 2017 09:28:58 GMT-0500 (CDT)
new Date("2017-07-12T14:28:58")
// Wed Jul 12 2017 14:28:58 GMT-0500 (CDT)
```

So if you try to actually change the date here, the values kind of jump around. This might be acceptable if we're just looking to test the switch from `date` to `datetime-local`, but it might be confusing for a new contributor going through the test fixtures.